src/historywork/DownloadBucketsWork.cpp

```diff
@@ -84,9 +85,16 @@ DownloadBucketsWork::yieldMoreWork()
             }
         }
     };
-
+    OnSuccessCallback successCb = [this, ft, hash]() {
+        auto bucketPath = ft.localPath_nogz();
+        auto b = this->mApp.getBucketManager().adoptFileAsBucket(
+            bucketPath, hexToBin256(hash),
+            /*objectsPut=*/0,
+            /*bytesPut=*/0);
+        this->mBuckets[hash] = b;
+    };
     auto w2 = std::make_shared<VerifyBucketWork>(
-        mApp, mBuckets, ft.localPath_nogz(), hexToBin256(hash), cb);
+        mApp, ft.localPath_nogz(), hexToBin256(hash), successCb, failureCb);
```

Comment:
not blocking here, but the better way (usable in other situations) would be to add a new `Work` adapter for a lambda so that it can be composed with `WorkSequence` and any other `Work` class.

Reply:
Marta merged the change I was asking about (this came up in a different PR), so you can use that new Work type to avoid adding ad-hoc callbacks like this

Reply:
Fixed.